auth-audits/src/main/java/org/apache/atlas/audit/destination/Log4JAuditDestination.java

```diff
@@ -75,12 +86,40 @@ public boolean log(AuditEventBase event) {
 		}
 
 		if (event != null) {
+            recordLogAttributes(event);
 			String eventStr = MiscUtil.stringify(event);
 			logJSON(eventStr);
+            clearLogAttributes(event);
 		}
 		return true;
 	}
 
+    private void recordLogAttributes(AuditEventBase event) {
+        if (event instanceof AuthzAuditEvent) {
+            MDC.put(AUTH_AUDIT_USER, ((AuthzAuditEvent) event).getUser());
+            MDC.put(AUTH_AUDIT_ACTION, ((AuthzAuditEvent) event).getAction());
+            MDC.put(AUTH_AUDIT_ENTITY_GUID, ((AuthzAuditEvent) event).getEntityGuid());
+            MDC.put(AUTH_AUDIT_POLICY_ID, ((AuthzAuditEvent) event).getPolicyId());
+            MDC.put(AUTH_AUDIT_RESOURCE, ((AuthzAuditEvent) event).getResourceType());
+            MDC.put(AUTH_AUDIT_RESULT, String.valueOf(((AuthzAuditEvent) event).getAccessResult()));
+			MDC.put(AUTH_AUDIT_CLIENT_IP, ((AuthzAuditEvent) event).getClientIP());
+			MDC.put(AUTH_AUDIT_AGENT, ((AuthzAuditEvent) event).getAgentId());
```

Comment:
There is no need to typeCast event to AuthzAuditEvent each time
can you create an AuthzAuditEvent object by typecasting an event & then use the object at all required lines